src/deprecated.jl

```diff
@@ -1,9 +1,79 @@
+using Base: @deprecate, depwarn
+
 function _dep_verbose_kw(verbose, preferred::Type, func::Symbol)
     if verbose !== nothing
-        Base.depwarn(
+        depwarn(
             "The `verbose` keyword is deprecated and now needs to be supplied " *
             "during the initialization of `$(nameof(preferred))`.",
             func,
         )
     end
 end
+
+@deprecate(
+    upload(ftp::FTP, local_name::AbstractString; kwargs...),
+    upload(ftp, local_name, local_name; kwargs...)
+)
+
+function upload(
+    ftp::FTP,
+    local_file_paths::Vector{<:AbstractString},
+    ftp_dir::AbstractString;
+    retry_callback::Function=(count, options) -> (count < 4, options),
+    retry_wait_seconds::Integer=5,
+    verbose=nothing,
+)
+    depwarn(
+        string(
+            "Uploading multiple files to an ftp_dir with a retry callback is deprecated. ",
+            "Use retry(upload, delays=fill(5, 4))",
+            "(ftp, local_path, remote_path; kwargs... ) ",
+            "in a loop of file_paths in the future."
+        ),
+        :upload
+    )
```

Comment:
I think this is clearer if we use multiple lines and show:
```julia
retry(delays=fill(5, 4)) do
    upload(ftp, local_path, remote_path; kwargs... )
end
```
I also think it's worth mentioning in the help for `upload` how users can implement retries

Reply:
makes sense. Added a bit of text in the readme, and changed the deprecation message.

Reply:
Although a bit strange this is easier to read:
```julia
depwarn(
    string(
        "Uploading multiple files to an ftp_dir with a retry callback is deprecated. ",
        """
        Use retry in a file path loop in the future:
        for single_file in file_paths
            retry(delays=fill(5, 4)) do
                upload(ftp, local_path, remote_path; kwargs... )
            end
        end
        """,
    ),
    :upload
)
```